File: android/src/main/java/com/wortise/ads/react/WortiseNativeAds.kt
package com.wortise.ads.react

object WortiseNativeAds {

    private val factories = mutableMapOf<String, WortiseNativeAdViewFactory>()


    @JvmStatic
    fun registerFactory(id: String, factory: WortiseNativeAdViewFactory) {
        factories[id] = factory
    }

    @JvmStatic
    fun unregisterFactory(id: String) {
        factories.remove(id)
    }


    internal fun getFactory(id: String): WortiseNativeAdViewFactory? = factories[id]
}

File: ios/RNWortiseSdk/WortiseNativeAds.swift
import Foundation
import WortiseSDK

@objc
public protocol WortiseNativeAdViewFactory {
    func createNativeAdView() -> WANativeAdView
}

@objc
public class WortiseNativeAds: NSObject {

    private static var factories = [String: WortiseNativeAdViewFactory]()


    @objc
    public static func registerFactory(_ id: String, factory: WortiseNativeAdViewFactory) {
        factories[id] = factory
    }

    @objc
    public static func unregisterFactory(_ id: String) {
        factories.removeValue(forKey: id)
    }


    static func getFactory(_ id: String) -> WortiseNativeAdViewFactory? {
        return factories[id]
    }
}
